internal/common/conversion/type_conversion.go

```diff
@@ -3,8 +3,18 @@ package conversion
 import (
 	"strings"
 	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
+func SafeValue[T any](v *T) T {
+	if v != nil {
+		return *v
+	}
+	emptyValue := new(T)
+	return *emptyValue
+}
+
 func SafeString(s *string) string {
```

Comment:
we can remove SafeString and use SafeValue isntead

Reply:
It is used in 11 files... Prefer doing it in a separate PR